src/pages/collaborations/collaborationManagementForm.jsx

```diff
@@ -111,12 +133,27 @@ export default function CollaborationManagementForm({ userData }) {
           style={{ marginBottom: '20px' }}
           loading={loading}
           onClick={async () => {
-            const successful = await establishCollaboration(
-              // need the await here. Otherwise, setShouldDisplay(true) below fires before this completes
-              user1,
-              user2,
-            );
-            setShouldDisplay(true);
+            if (
+              !collaborationAlreadyExists(
+                existingCollaborations,
+                user1,
+                user2,
+              )
+            ) {
+              await establishCollaboration(
+                // need the await here. Otherwise, setShouldDisplay(true) below fires before this completes
+                user1,
+                user2,
+              );
+              setShouldDisplay(true);
+            } else {
+              setError(
+                intl.formatMessage({
+                  id: 'COLLABORATION_ALREADY_EXISTS',
+                }),
+              );
+              setShouldDisplay(true);
```

Comment:
since `setShouldDisplay(true)` is called in the `if` and `else` clauses, it might make more sense to just put it underneath the clause